subjects/ai/numpy/README.md

```diff
@@ -209,60 +248,95 @@ grades[[1,2,5,7], [0,0,0,0]] = np.nan
 print(grades)
 ```
 
+This code returns:
+
+```console
+[[ 7.  1.]
+ [nan  2.]
+ [nan  8.]
+ [ 9.  3.]
+ [ 8.  9.]
+ [nan  2.]
+ [ 8.  2.]
+ [nan  6.]
+ [ 9.  2.]
+ [ 8.  5.]]
+```
+
+1. Using `np.where`, create a third column that takes the grade of the first exam if available; otherwise, it uses the grade from the second exam. Add this column as the third column of the array.
+
+**Using a for loop or if/else statement is not allowed in this exercise.**
+
+Expected output:
+
+```console
+[[ 7.  1.  7.]
+ [nan  2.  2.]
+ [nan  8.  8.]
+ [ 9.  3.  9.]
+ [ 8.  9.  8.]
+ [nan  2.  2.]
+ [ 8.  2.  8.]
+ [nan  6.  6.]
+ [ 9.  2.  9.]
+ [ 8.  5.  8.]]
+```
+
 ---
 
 ---
 
-# Exercise 8: Wine
+## Exercise 8: Wine
 
-The goal of this exercise is to learn to perform a basic data analysis on real data using NumPy.
+The goal of this exercise is to perform fundamental data analysis on real data using NumPy.
 
-The data set that will be used for this exercise is the red wine data set.
+The dataset chosen for this task is the [red wine dataset](https://archive.ics.uci.edu/ml/datasets/wine+quality)
```

Comment:
this could be linked to the data in the `./data/` directory to avoid having broken links in the future.